solanaj/src/main/java/org/p2p/solanaj/kits/transaction/parser/ConfirmedTransactionRootParser.kt

```diff
@@ -0,0 +1,201 @@
+package org.p2p.solanaj.kits.transaction.parser
+
+import org.p2p.solanaj.kits.transaction.BurnOrMintDetails
+import org.p2p.solanaj.kits.transaction.CloseAccountDetails
+import org.p2p.solanaj.kits.transaction.CreateAccountDetails
+import org.p2p.solanaj.kits.transaction.TransactionDetails
+import org.p2p.solanaj.kits.transaction.TransferDetails
+import org.p2p.solanaj.kits.transaction.UnknownDetails
+import org.p2p.solanaj.kits.transaction.network.ConfirmedTransactionRootResponse
+import org.p2p.solanaj.kits.transaction.network.meta.InstructionInfoDetailsResponse
+import org.p2p.solanaj.kits.transaction.network.meta.InstructionResponse
+import org.p2p.solanaj.programs.SystemProgram.SPL_TOKEN_PROGRAM_ID
+import java.math.BigInteger
+
+// todo: Parser should be refactored and optimized
+// Get some parsing info from here
+// https://github.com/p2p-org/solana-swift/blob/main/Sources/SolanaSwift/Helpers/TransactionParser.swift#L74-L86
+class ConfirmedTransactionRootParser(
+    private val orcaSwapInstructionParser: OrcaSwapInstructionParser,
+    private val serumSwapInstructionParser: SerumSwapInstructionParser
+) {
+
+    fun parse(
+        transactionRoot: ConfirmedTransactionRootResponse,
+        onErrorLogger: (Throwable) -> Unit
+    ): List<TransactionDetails> {
+        val signature = transactionRoot.transaction?.getTransactionId() ?: return emptyList()
+
+        return when {
+            orcaSwapInstructionParser.isTransactionContainsOrcaSwap(transactionRoot) -> {
+                listOfNotNull(
+                    orcaSwapInstructionParser.parse(signature, transactionRoot)
+                        .onFailure(onErrorLogger)
+                        .getOrNull()
+                )
+            }
+            serumSwapInstructionParser.isTransactionContainsSerumSwap(transactionRoot) -> {
+                listOfNotNull(
+                    serumSwapInstructionParser.parse(signature, transactionRoot)
+                        .onFailure(onErrorLogger)
+                        .getOrNull()
+                )
+            }
+            else -> {
+                parseTransaction(transactionRoot, signature)
+            }
+        }
+    }
+
+    private fun parseTransaction(
+        transactionRoot: ConfirmedTransactionRootResponse,
+        signature: String,
+    ): List<TransactionDetails> {
+        return transactionRoot.transaction
+            ?.message
+            ?.instructions
+            ?.mapNotNull { parseInstructionByType(signature, transactionRoot, it) }
+            ?.toMutableList()
+            .orEmpty()
+    }
+
+    private fun parseInstructionByType(
+        signature: String,
+        transactionRoot: ConfirmedTransactionRootResponse,
+        parsedInstruction: InstructionResponse
+    ): TransactionDetails? {
+        val parsedInfo = parsedInstruction.parsed
+        return when (parsedInfo?.type) {
+            "burnChecked" -> {
```

Comment:
maybe we will move all names in enum ?

Reply:
they are used only here, so constants I think is good. 
or maybe I can write a custom converter str -> enum to remove strings from `type` field.
whatcha think?